File: 2017-18-1/af3/alkfejl-angular/src/app/classes/todo.ts
import { User } from './user';

export class Todo {
    private _id: number;
    private _text: string;
    private _user: User;

    public constructor(id: number, text: string, user: User) {
        this._id = id;
        this._text = text;
        this._user = user;
    }

    public get id(): number {
        return this._id;
    }

    public get text(): string {
        return this._text;
    }

    public get user(): User {
        return this._user;
    }

    public toString(): string {
        return this.text;
    }

}

File: 2017-18-1/af3/alkfejl-rest/src/main/java/com/elte/alkfejlrest/entity/Todo.java
package com.elte.alkfejlrest.entity;

import groovy.transform.EqualsAndHashCode;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Integer id;
    
    @Column(nullable=false)
    private String text;
    
    @JoinColumn
    @ManyToOne(targetEntity = User.class)
    private User user;
}
